examples/vae/train.py

```diff
@@ -143,16 +143,17 @@ def loss_fn(params):
 
 @jax.jit
 def eval(params, images, z, z_rng):
-  recon_images, mean, logvar = model().apply({'params': params}, images, z_rng)
+  def eval_model(vae):
```

Comment:
Curious why you ported this example to use `nn.apply`?

Reply:
It's not essential but I think it's nice to have at least one usage in the examples. Here we avoid passing custom methods and a double apply

Reply:
Ah yes I now fully agree that this is a good idea. I always disliked the `method` argument to `apply`... (And if we really want to we /could/ get rid of it now... But not sure)